version: "3.3"
services:
  rabbitmq:
    image: rabbitmq:3-management-alpine
    hostname: rabbit
    container_name: rabbitmq
    ports:
      - 15672:15672
      - 5672:5672
    volumes:
      - ./.rabbitmq_data:/var/lib/rabbitmq/mnesia
    environment:
      - RABBITMQ_DEFAULT_USER=fvbnnbsb
      - RABBITMQ_DEFAULT_PASS=Te-b67XCoOoLkO8vyGILAFnh7xzSMJP2
    healthcheck:
      test: [ "CMD", "rabbitmqctl", "status" ]
      interval: 30s
      timeout: 10s
      retries: 5
    networks:
      - app_network

  rtmp:
    build: ./apps/rtmp
    restart: always
    ports:
      - "1935:1935"
      - "8080:8080"
    container_name: rtmp_server
    volumes:
      - ./data:/tmp/hls
    networks:
      - app_network

  auth:
    build: ./apps/auth
    restart: always
    ports:
      - "8000:8000"
    container_name: auth_server
    environment:
      - PORT=8000
    networks:
      - app_network
  
  comments-service:
    build: ./apps/comments-service
    restart: always
    ports:
      - "3007:3007"
    container_name: comments-service
    environment:
      - PORT=3007
    depends_on:
            rabbitmq:
                condition: service_healthy
    networks:
      - app_network

  filter-service:
    build: ./apps/filter-service
    restart: always
    ports:
      - "3010:3010"
    container_name: filter-service
    environment:
      - PORT=3010
    depends_on:
            rabbitmq:
                condition: service_healthy
    networks:
      - app_network

networks:
  app_network:
    external: true
